nixos/modules/services/x11/extra-layouts.nix

```diff
@@ -0,0 +1,156 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  layouts = config.services.xserver.extraLayouts;
+
+  layoutOpts = {
+    options = {
+      description = mkOption {
+        type = types.str;
+        description = "A short description of the layout.";
+      };
+
+      languages = mkOption {
+        type = types.listOf types.str;
+        description =
+        ''
+          A list of languages provided by the layout.
+          (Use ISO 639-2 codes, for example: "eng" for english)
+        '';
+      };
+
+      layoutFile = mkOption {
+        type = types.path;
+        description = "The path to the layout file.";
+      };
```

Comment:
This indicates strongly that this option should have a good example and a better description with some links to docs or so.

Reply:
And @laMudri pointed out that if it's such a symbols file, this option should be called `symbolsFile` instead. I didn't realize this earlier.

Reply:
Also, the build should probably fail somehow if the file given has the wrong format

Reply:
More thoroughly, the motivation for including the word `symbols` is:
1) Future work might make the files in `compat/`, `geometry/`, `keycodes/`, and `types/` customisable in this options block. So the file in `symbols/` is not necessarily the whole layout. (I'm not sure what these files do, but in principle at least they might be customisable. Who knows what the future might bring?)
2) `symbols` is xkb jargon, which might be helpful to see for a user who is trying to work out what to put in this file. As @Infinisil's test case showed, it's not obvious that `layoutFile` is just for symbols, and not for other stuff too.

Reply:
I agree. Including other sections shouldn't be much different, I'll give it a try.
Anyway, I updated the description: better?